src/com/hms/flexyosisoftconnector/PayloadBuilder.java

```diff
@@ -0,0 +1,382 @@
+package com.hms.flexyosisoftconnector;
+
+import com.ewon.ewonitf.SysControlBlock;
+import com.hms_networks.americas.sc.datapoint.DataPoint;
+import com.hms_networks.americas.sc.datapoint.DataType;
+import com.hms_networks.americas.sc.json.JSONException;
+import com.hms_networks.americas.sc.logging.Logger;
+import com.hms_networks.americas.sc.taginfo.TagInfo;
+import com.hms_networks.americas.sc.taginfo.TagInfoManager;
+import com.hms_networks.americas.sc.taginfo.TagType;
+
+import java.util.ArrayList;
+
+/**
+ * Class to build up a JSON payload in segments of strings.
+ *
+ * @author HMS Networks Inc. Solution Center
+ */
+public class PayloadBuilder {
+
+  /**
+   * OSIsoft server's ip address appende used in the PIWEBAPI URL
+   */
+  private static String targetURL;
+
+  /**
+   * OSIsoft server's login credentials
+   */
+  private static String authCredentials;
+
+  /**
+   * array of tag web id's
+   */
+  private static String tagWebIdList[];
+
+  /**
+   * Index offset used to find the tagWebIDList index from the tag ID.
+   */
+  private static int tagWebIdListIndexOffset;
+
+  /**
+   * Initialize an array of tag web ID's for legacy function web ID lookup.
+   */
+  public static void initWebIdList() {
+    // setup an indicted array of web Id's
+    int lowestSeen = TagInfoManager.getLowestTagIdSeen();
+    int highestSeen = TagInfoManager.getHighestTagIdSeen();
+    tagWebIdListIndexOffset = lowestSeen;
+    int tagListArraySize = highestSeen - lowestSeen + 1;
+
+    if (tagListArraySize == -1) {
+      tagListArraySize = 1;
+    }
+    tagWebIdList = new String[tagListArraySize];
+  }
+
+  /**
+   * Set the web idea to the retrieved web id from initializing a tag.
+   *
+   * @param tagId tag's ID
+   * @param webID tag's webID
+   */
+  public static void setTagWebId(int tagId, String webID) {
+    int index = tagId - tagWebIdListIndexOffset;
+    tagWebIdList[index] = webID;
+  }
+
+  /**
+   * Configure the login credentials. This function is for the legacy OSIsoft formats.
+   *
+   * @param login base 64 encoded username and password for login credentials.
+   */
+  public static void setAuthCredentials(String login) {
+    authCredentials = login;
+  }
+
+  /**
+   * Configure where payloads get sent. This function is for the legacy OSIsoft formats.
+   *
+   * @param serverIP ip address of OSIsoft server.
+   */
+  public static void setTargetURL(String serverIP) {
+    targetURL = "https://" + serverIP + "/piwebapi/";
+  }
```

Comment:
Why are you storing copies of these things. These already are stored in memory within the connector. 

Also Payload builder is probably not the most logical place to have these.

Reply:
addPointOldFormat for the old format needs that info every time a new point is built. it's just in the JSON that gets returned and it made sense at the time to just store a copy in the class instead of constantly grabbing it for every new data point added. I could probably change that around if needed. there is also a copy in the osisoft server because for anything OMF, that information is only used when sending messages which is all handled in the OSIsoft server class

Reply:
same with the auth credentials and tag web id, needed for every new point to add to the json

Reply:
right but you should be able to access them directly from the config class

Reply:
true, I'll switch that.

Reply:
I should make the config class static for this behavior to be a little nicer to use. right now it is not which is why this ended up like this

Reply:
okay